backbone/src/main/java/org/researchstack/backbone/result/TaskResult.java

```diff
@@ -27,10 +34,24 @@
     // unimplemented but exists in RK, implement or delete if not needed
     private Uri outputDirectory;
 
+    /* Default identifier for serilization/deserialization */
+    TaskResult() {
+        super();
+        this.results = new LinkedHashMap<>();
```

Comment:
LinkedHashMaps retain order of insertion, which the iOS algorithms use sometimes.  So I figured it would be best to keep the results ordered if we ever iterate them.